hazelcast-client-legacy/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -235,24 +237,36 @@ protected void loadAllInternal(boolean replaceExistingValues, List<Data> dataKey
     }
 
     @Override
-    protected MapEntries getAllInternal(List<Data> keys, Map<K, V> result) {
-        Iterator<Data> iterator = keys.iterator();
-        while (iterator.hasNext()) {
-            Data key = iterator.next();
-            Object cached = nearCache.get(key);
-            if (cached != null && NULL_OBJECT != cached) {
-                result.put((K) toObject(key), (V) cached);
-                iterator.remove();
+    protected List<MapEntries> getAllInternal(Map<Integer, List<Data>> partitionToKeyData, Map<K, V> result) {
+        List<Integer> partitionsWithCachedEntries = new ArrayList<Integer>(partitionToKeyData.size());
+        for (Entry<Integer, List<Data>> partitionKeyEntry : partitionToKeyData.entrySet()) {
+            List<Data> keyList = partitionKeyEntry.getValue();
+            Iterator<Data> iterator = keyList.iterator();
+            while (iterator.hasNext()) {
+                Data key = iterator.next();
+                Object cached = nearCache.get(key);
+                if (cached != null && NULL_OBJECT != cached) {
+                    result.put((K) toObject(key), (V) cached);
+                    iterator.remove();
+                }
+            }
+            if (keyList.isEmpty()){
+                partitionsWithCachedEntries.add(partitionKeyEntry.getKey());
             }
         }
-
-        MapEntries entries = super.getAllInternal(keys, result);
-
-        for (Entry<Data, Data> entry : entries.entries()) {
-            nearCache.put(entry.getKey(), entry.getValue());
+        
+        for (Integer partitionId : partitionsWithCachedEntries) {
+            partitionToKeyData.remove(partitionId);
+        }
+        
+        List<MapEntries> responses = super.getAllInternal(partitionToKeyData, result);
+        for (MapEntries entries : responses) {
+            for (Entry<Data, Data> entry : entries.entries()) {
+                nearCache.put(entry.getKey(), entry.getValue());
+            }
         }
         //TODO: This returned value is not be used.
-        return entries;
+        return responses;
     }
```

Comment:
we are potentially retrieving items here which are already cached if they belong to a partition in which some items are cached and some not. What is the reason for changing this method from it's default behaviour (only retrieve non-cached items)